### DISCUSSION TITLE
Error saving Site Settings, new install on IIS web server

### DISCUSSION THREAD

**Discussion by LuisVilaplana** [oqtane/oqtane.framework#3264] [General] [answered]
Hi, I'm getting this error when saving Site Setings on a new installation on public IIS:

Type name: Oqtane.Modules.Admin.Site.Index, Oqtane.Client, Version=4.0.3.0, Culture=neutral, PublicKeyToken=null

Page: Site Settings

Url: https://www.prosoftpos.net/admin/site

Template: Error Saving Site {SiteId} {Error}

Message: Error Saving Site 1 Object reference not set to an instance of an object.

Exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at Oqtane.Modules.Admin.Site.Index.SaveSite() in C:\Source\Projects\oqtane.framework\Oqtane.Client\Modules\Admin\Site\Index.razor:line 604

Properties: {"SiteId":1,"Error":"Object reference not set to an instance of an object."}
-----------------
I'm not having this error using source code on Visual Studio (localhost). This installation is from downloaded file from Github

Any idea how to fix this error?

Thanks

**Comment by sbwalker** [comment]
@LuisVilaplana I am not able to reproduce this problem. Can you please provide information about the hosting model, database engine, etc... that you are using in your environment.

**Reply by LuisVilaplana** [reply]
Hello, this is my setup:

Hosting model: Server
Database: SQL Server Express
Web server: IIS

This is a completely new installation, from the files downloaded from Github (Windows 64bits). I followed all the official instructions to configure IIS

**Reply by sbwalker** [reply]
Environmental problems on IIS are usually caused by:

- WebSocket Protocol is disabled 
- WebDAV Publishing is enabled

Can you please include screen shots of these settings on your server

**Reply by LuisVilaplana** [reply]
I'm trying to disable both options for this site only because I don't want to cause problems to other sites or the Plesk panel itself that are hosted on this same server. I have tried to disable them in Web.Config but some errors are generated. Do you have an example of how that web.config should look like?

Thanks

**Reply by leigh-pointer** [reply]
Search in discussion for WebDAV.  I have addressed this a few times

**Reply by LuisVilaplana** [reply]
Ok, I will, thanks

**Reply by LuisVilaplana** [reply]
Ok, I tried configuring WebDav and WebSockets in IIS but the problem persists. This is my Web.Config file:

![image](https://github.com/oqtane/oqtane.framework/assets/103559699/5d1951ba-1be4-4762-9e2e-7209cc2c7a31)

**Comment by LuisVilaplana** [comment, marked as answer]
Ok, I was able to fix the error setting my Web.Config file like this:

![image](https://github.com/oqtane/oqtane.framework/assets/103559699/8735015c-ce6d-4734-8e2c-1ecc3a48df27)

Thanks a lot
